### PR TITLE
Missing which and too fast failures

### PR DESCRIPTION
**Pull request by hhorak**
We don't have `which` in all containers. Also, in some cases when, when tests fail before the first container is created, the cleanup can fail by trying to remote empty string.

### CODE REVIEW

File: test-lib.sh [hunk 1]
```diff
@@ -27,6 +27,7 @@ EXPECTED_EXIT_CODE=0
 function ct_cleanup() {
   ct_show_resources
   for cid_file in "$CID_FILE_DIR"/* ; do
+    [ -f "$cid_file" ] || break
```
Comment (phracek) [inline test-lib.sh:30]: If dir `$CID_FILE_DIR` contains more files then only first is checked. What about `continue`?

**Review by phracek — CHANGES_REQUESTED** [review]

File: test-lib.sh (hunk 1 above)
Comment (hhorak) [reply test-lib.sh:30]: good idea.

Comment (hhorak) [reply test-lib.sh]: fixed.

**Review by phracek — APPROVED** [review]
LGTM. Thanks for the correction.

**Comment by phracek** [comment]
[test]
